### DISCUSSION TITLE
jdk 17 compatibility

### DISCUSSION THREAD

**Discussion by Czytx** [dropwizard/dropwizard#6394] [Q&A]
Is there a minimum version of DW to get it to work on OpenJDK 17?

**Comment by zUniQueX** [comment]
Hi @Czytx. We run the framework's unit tests against JDK 17 for all currently maintained versions (2.0.x onwards).

**Reply by joschi** [reply]
Additionally to what @zUniQueX wrote, I'd recommend going with Dropwizard 4.0.0 (latest beta) on Java 17 when starting a new project.

It will soon go GA, so don't start with Dropwizard 2.x. 😄

**Reply by Czytx** [reply]
I'm trying to migrate our projects to Java 17. Our dropWizard version is 2.0.19, jersey version is 2.33. But I find this :https://github.com/dropwizard/dropwizard/issues/4347.
Does that mean that I will migrate dropWizard version to 2.1.x ?

**Reply by joschi** [reply]
> Does that mean that I will migrate dropWizard version to 2.1.x ?

Given that end-of-life of Dropwizard 2.0.x is in less than 1 month (see https://github.com/dropwizard/dropwizard/discussions/6324), I would definitely recommend migrating to Dropwizard 2.1.x or even better 3.0.x (or 4.0.x if you want to invest a bit more time).
